lib/reek/configuration/app_configuration.rb

```diff
@@ -90,10 +90,11 @@ def self.new(*)
       #
       # @param source_via [String] - the source of the code inspected
       #
-      # @return [Hash] the directory directive for the source or, if there is
-      # none, the default directive
+      # @return [Hash] the directory directive for the source with the default directive
+      #                reverse-merged into it.
       def directive_for(source_via)
-        directory_directives.directive_for(source_via) || default_directive
+        hit = directory_directives.directive_for(source_via)
+        hit ? default_directive.merge(hit) : default_directive
```

Comment:
Well, this is only because we have a primitive obsession and our directives are hashes. :)

If this method actually returned a directive for a directory, then there could be a null directive for a directory that has no directives. But it doesn’t return a directive for a directory, it returns a hash (or a nil).

Refactoring this to actually use directory directives is definitely out of scope of this PR (and I’m not saying we should do this; hashes served us good enough so far) – and if we are to stick to hashes then I agree with you; so, merging!